packages/mobile/src/screens/favorites-screen/PlaylistsTab.tsx

```diff
@@ -35,6 +36,16 @@ export const PlaylistsTab = () => {
   const isOfflineModeEnabled = useIsOfflineModeEnabled()
   const isReachable = useSelector(getIsReachable)
   const isDoneLoadingFromDisk = useSelector(getIsDoneLoadingFromDisk)
+  const offlineDownloadStatus = useProxySelector(
```

Comment:
based on convos with @sliptype, i think the best setup for playlistsTab would be for its only job to be getting collectionIds that we want to render, online or offline, and pass that into the card list, and then each playlist card would be in charge of getting the tracks, online or offline, which we can then derive the count etc. @nicoback curious how you feel about that as the paradigm here? would it be too much of a refactor?

Reply:
it'd be a bit of a refactor just because collection card list is used in several places elsewhere in the code (like 10 places), not just the Playlists/Albums tabs. But I'm down to do it, it wouldn't be too bad. cc @sliptype

Reply:
Like def agree that is the way to go lol. This code is pretty suboptimal.

Reply:
Ooh gotcha. Yeah thinking about it tho, i think it wouldn't take too much time. @sliptype if you aren't working on this, i can pick it up first thing tomorrow and timebox to half an hour or so

Reply:
I haven't gotten a chance to look at the CardList yet, got a couple of other refactor to knock out first. Yeah @nicoback definitely feel free to look at it if you want, but maybe let's see if there is anything else more pressing on the backlog first

Reply:
oh yeah there are definitely more pressing things. ill leave this open and come back to it after hitting more high pri items then